pex/pex_bootstrapper.py

```diff
@@ -72,33 +72,40 @@ def _iter_interpreters():
         # type: () -> Iterator[InterpreterOrError]
         seen = set()
 
-        paths = None  # type: Optional[MutableSet[str]]
-        if path:
-            paths = OrderedSet(os.path.realpath(p) for p in path.split(os.pathsep))
+        normalized_paths = (
+            OrderedSet(os.path.realpath(p) for p in path.split(os.pathsep)) if path else None
+        )
 
+        # Check if the current interpreter is valid. If `path` was specified, the current
+        # interpreter must be included in the value; otherwise, we always yield the interpreter.
```

Comment:
The comment you deleted below has important information now missing here, namely that we yield the current interpreter *before all others*, so that it's preferred.

Reply:
I'll add that back. 

FYI the part I wanted to remove was ` # We may have been invoked with a specific interpreter` as I believe that's misleading; regardless of the current interp is via shebang discovery or specific interpreter, we should consider it.

Reply:
Yes, and we are always invoked with a specific interpreter, in some sense...